generator/internal/http_option_utils.h

```diff
@@ -31,13 +31,20 @@ struct HttpSimpleInfo {
 };
 
 struct HttpExtensionInfo {
+  struct RestPathPiece {
+    RestPathPiece(std::string piece, bool format_as_accessor)
+        : piece(std::move(piece)), format_as_accessor(format_as_accessor) {}
+    std::string piece;
+    bool format_as_accessor;
```

Comment:
This smells like a type code in disguise.  Maybe want you want is:

```
using RestPathPiece = std::function<std::string(google::protobuf::MethodDescriptor const&)>;
...
std::vector<RestPath> rest_path;
```

and then you just initialize them to either:

```
info.rest_path.emplace_back([piece = foo.first](google::protobuf::MethodDescriptor const& method) {
     return absl::StrFormat("request.%s()", FormatFieldAccessorCall(method, p.piece));
});
```

or to:

```
info.rest_path.emplace_back([piece = foo.first](google::protobuf::MethodDescriptor const&) {
     return absl::StrFormat("\"%s\"", p.piece);
});
```